### PR TITLE
Add jsdoc @type for autocompletion

### PR DESCRIPTION
**Pull request by mhf-ir**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: docs/TypeScript.md [hunk 1]
```diff
@@ -44,6 +44,25 @@ server.get('/ping', opts, (request, reply) => {
 })
 ```
 
+<a id="jsdoc"></a>
+### JSDoc
```
[the remark this answers is not in the document]
Comment (mhf-ir) [reply docs/TypeScript.md]: Your right maybe in wrong place in documentation, but autocompletion is one of favorite feature of typescript. Many developers like me looking for it. 
Problem is when you're separate js files editors confused about type of arguments jsdoc will be great.

[the remark this answers is not in the document]
Comment (mhf-ir) [reply docs/TypeScript.md]: May be in documentation we need something like boilerplate or cli stuff then this will be there :confused:

**Comment by cemremengu** [comment]
While I think this is helpful, I think it belongs more to `JSDoc` or `vscode` documentation since it is not related to Fastify at all.

**Comment by cemremengu** [comment]
Closing this one since I think it is settled, feel free to reopen if further discussion needed

**Pull request closed without merging**